core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -211,6 +211,32 @@ private[spark] class AppStatusListener(
           update(rdd, now)
         }
       }
+      // remove partition information
+      liveRDDs.values.foreach { rdd =>
+        rdd.getPartitions.values.foreach { partition =>
+          partition.executors.foreach { exec =>
+            if (exec.equals(event.executorId)) {
+              if (partition.executors.length == 1) {
+                rdd.removePartition(partition.blockName)
```

Comment:
You're removing a partition while iterating over the partition map. That's generally not safe.